### PR TITLE
Add Pkl `codegen` related macros

### PR DESCRIPTION
**Pull request by shs96c**
This commit contains the code requires for `pkl_codegen_java_toolchain`, `pkl_config_java_library`, and `pkl_config_src`.

This PR also bumps some bzlmod modules to the latest public releases. This was required to get the docs generation working with the version of `rules_java` we use.

### CODE REVIEW

File: pkl/private/pkl_codegen.bzl [hunk 1]
```diff
@@ -0,0 +1,147 @@
+"""
+Implementation for 'pkl_config_src' and 'pkl_config_java_library' macros.
+"""
+
+load("@rules_jvm_external//:defs.bzl", "artifact")
+
+def _to_short_path(f, expander):
```
Comment (KushalP) [inline pkl/private/pkl_codegen.bzl:7]:
```suggestion
def _to_short_path(f, _expander):
```

As per the build failure:

```
./pkl/private/pkl_codegen.bzl:7: unused-variable: Variable "expander" is unused. Please remove it. (https://github.com/bazelbuild/buildtools/blob/master/WARNINGS.md#unused-variable)
```

Comment (shs96c) [reply pkl/private/pkl_codegen.bzl]: Done

**Review by KushalP — APPROVED** [review]

File: tests/pkl_codegen_java/TestPklCodegen.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+public class TestPklCodegen {
+    @Test
+    public void verifyReadingPklFiles() {
+        ConfigEvaluator evaluator = ConfigEvaluator.preconfigured();
+        Config config = evaluator.evaluate(ModuleSource.modulePath("srcs/litter.pkl"));
+
+        org.Dogs.Litter actual = config.get("puppies").as(JavaType.of(Dogs.Litter.class));
+        org.Dogs.Litter expected = new org.Dogs.Litter(List.of(new Dogs.Dog("Bruno"), new Dogs.Dog("Bruce")));
+
+
```
Comment (KushalP) [inline tests/pkl_codegen_java/TestPklCodegen.java:24]: We may want to integrate one of the formatter rule sets to take of stuff like this and anything else that may have been missed with these Java files.

Comment (shs96c) [reply tests/pkl_codegen_java/TestPklCodegen.java]: I've removed this one.

**Pull request merged**